### PR TITLE
Better doc and bindings for gmf search

### PR DESCRIPTION
**Pull request by ger-benjamin**
GSGMF-385

Missing "?" in bindings create a warning in the console, and a problem on usage.

### CODE REVIEW

File: contribs/gmf/src/search/component.js [hunk 1]
```diff
@@ -420,6 +420,7 @@ gmf.search.component.SearchController_ = class {
     const gettextCatalog = this.gettextCatalog_;
     this.clearButton = this.clearButton !== false;
     this.colorChooser = this.colorChooser === true;
+    this.delay = parseInt(this.delay, 10);
```
Comment (sbrunner) [inline contribs/gmf/src/search/component.js]: @gberaudo is this parseInt requied?

File: contribs/gmf/src/search/component.js [hunk 2]
```diff
@@ -321,7 +321,7 @@ gmf.search.component.SearchController_ = class {
      * @type {number}
      * @export
      */
-    this.delay = parseInt(this.scope_['delay'], 10) || 50;
+    this.delay = 50;
```
Comment (gberaudo) [inline contribs/gmf/src/search/component.js]: You should not assign to `this.delay` in the constructor. The value will be lost.

File: contribs/gmf/src/search/component.js (hunk 1 above)
Comment (gberaudo) [reply contribs/gmf/src/search/component.js]: @sbrunner, maybe the value is sometime `undefined`?
I would expect the following to be more correct:
```
if (this.delay === undefined) {
  this.delay = 50;
}
```

File: contribs/gmf/src/search/component.js [hunk 3]
```diff
@@ -134,14 +134,14 @@ function gmfSearchTemplateUrl($element, $attrs, gmfSearchTemplateUrl) {
...
- * @htmlAttribute {boolean} gmf-search-clearbutton The clear button.
- * @htmlAttribute {boolean} gmf-search-delay The bloodhound request delay.
- * @htmlAttribute {boolean} gmf-search-colorchooser Whether to let the user
- *      change the style of the feature on the map. Default is false.
  * @htmlAttribute {ngeox.SearchComponentListeners} gmf-search-listeners
  *      The listeners.
- * @htmlAttribute {number} gmf-search-maxzoom The maximum zoom we will zoom on result, default is 16.
- * @htmlAttribute {function} gmf-search-on-init Optional function called when the component is initialized.
+ * @htmlAttribute {boolean=} gmf-search-clearbutton Optional clear button in the input search. Default to true.
+ * @htmlAttribute {boolean=} gmf-search-delay Optional bloodhound request delay in ms. Default to 50 ms.
```
Comment (gberaudo) [inline contribs/gmf/src/search/component.js]: shouldn't it be a number?

File: contribs/gmf/src/search/component.js (hunk 2 above)
Comment (ger-benjamin) [reply contribs/gmf/src/search/component.js]: no

**Comment by ger-benjamin** [comment]
Done

**Pull request merged**